zebra/zebra_nhg.c

```diff
@@ -2048,11 +2048,12 @@ static int nexthop_active(struct nexthop *nexthop, struct nhg_hash_entry *nhe,
 		return 1;
 	}
 
-	if (top &&
-	    ((top->family == AF_INET && top->prefixlen == 32
-	      && nexthop->gate.ipv4.s_addr == top->u.prefix4.s_addr)
-	     || (top->family == AF_INET6 && top->prefixlen == 128
-		 && memcmp(&nexthop->gate.ipv6, &top->u.prefix6, 16) == 0))) {
+	if (top
+	    && ((top->family == AF_INET && top->prefixlen == IPV4_MAX_BITLEN
+		 && nexthop->gate.ipv4.s_addr == top->u.prefix4.s_addr)
+		|| (top->family == AF_INET6 && top->prefixlen == IPV6_MAX_BITLEN
+		    && memcmp(&nexthop->gate.ipv6, &top->u.prefix6, 16)
```

Comment:
MAX_BYTELEN ? (since you're on a tear here...)

Reply:
Right, I'll check for this as well, thanks.